V3

AMazur2 opened the pull request:
Added roulette selection

AMazur2 committed:
Roulette selection added

AMazur2 committed:
Roulette selection added

AMazur2 committed:
Changed rouletteParentSelector in order to make it work wile minimalization

HasuNoHana commented on src/evolutionaryAlgorithm/EvolutionSimulator.py:
```diff
@@ -52,6 +53,14 @@ def run(self):
     def initialize_population(self) -> List[IndividualInterface]:
         return self.initializator.initialize()
 
+    def markPopulation(self, population: List[IndividualInterface]) -> List[float]:
+        fitness = []
+        for el in population:
+            i = 0
+        #    i = ocena(el)
+            fitness.append(i)
+        return fitness
+
```
ocena powinna być atrybutem osobnika a nie osobnąlistą floatów

AMazur2 replied:
to dodałem prowizorycznie żeby było pokazane że musimy to zrobić

HasuNoHana replied:
Po prostu dodaj atrybut do reprezentacji osobnika, robienie prowizorycznie powoduje chaos.

HasuNoHana commented on src/evolutionaryAlgorithm/SimulationComponents/ParentSelector/ParentSelectorInterface.py:
```diff
@@ -7,5 +7,5 @@
     
     @abstractmethod
-    def marry(self, population: List[IndividualInterface]):
+    def marry(self, population: List[IndividualInterface], fitness: List[float]):
```
fitness powinno być atrybutem osobnika

HasuNoHana commented on src/evolutionaryAlgorithm/SimulationComponents/ParentSelector/impl/rouletteParentSelector.py:
```diff
@@ -0,0 +1,46 @@
+    def __init__(self):
+        pass
+
+    def marry(self, Population: List[IndividualInterface], fitness: List[float]):
```
Nazwa jest mało mówiąca, proponuję get parents bo wiadomo co zwraca. Ta metoda powinna być abstrakcyjną metodą w interfacie `ParentSelectorInterface` . Jak wyżej poowinna przyjmowac tylko Population

AMazur2 replied:
ok, tak też będzie

HasuNoHana commented on src/evolutionaryAlgorithm/SimulationComponents/ParentSelector/impl/rouletteParentSelector.py:
```diff
@@ -0,0 +1,46 @@
+                k = self.chooseOne(fitnessPrim, sum)
+                parents.append(Population[k])
+            marriages.append(parents)
+        return marriages
```
Jest to długa metoda która implementuje logikę, przydałoby się zrobić test jesdnostkowy żeby sprawdzić czy działa dla chociaż prostych przypadków

HasuNoHana commented on src/evolutionaryAlgorithm/SimulationComponents/ParentSelector/impl/rouletteParentSelector.py:
```diff
@@ -0,0 +1,46 @@
+            marriages.append(parents)
+        return marriages
+
+    def chooseOne(self, fitness: List[float], maximum: float):
```
co ta funkcja robi, jak wybiera jednego osobnika?

AMazur2 replied:
no wybiera jednego osobnika bo tak to był problem z break'iem więc to chooseOne to można sobie wyobrazić jako jedno odpalenie ruletki i ona zwraca nam jakiś numer - indeks osobnika

HasuNoHana replied:
A jak to działa? na podstawie fitness czy każdy osobnik ma taką samą szansę czy co? Bo nazwa nic nie  mówi albo dopisz komentarz

HasuNoHana commented on test/evolutionaryAlgorithm/test_EvolutionSimulatorBuilder.py:
```diff
@@ -27,7 +27,7 @@ def test_create_evolution_simulator_from_dict():
                         "Probability" : 0.5
                     }},
-        "ParentSelector": {"Type": "simpleParentSelector"},
+        "ParentSelector": {"Type": "rouletteParentSelector"},
```
Jeśli chcemy kożystać z obu selektorów należy napisać drugi test(żeby przetestować oba)

AMazur2 replied:
Nie powinniśmy raczej z obu ja bym zostawił ruletkowe

HasuNoHana replied:
Okej, to kod musi się kompilować z rulete parent selector a obecnie rulete parent factory zwraca not implemented error

AMazur2 committed:
Mutator changed - no probability needed

AMazur2 committed:
Mutator changed - no probability needed

AMazur2 committed:
Mutator changed - no probability needed

AMazur2 committed:
Changed some issues

AMazur2 committed:
Test mostly works

AMazur2 committed:
rouletteParentSelector fixed
added test of rouletteParentSelector

AMazur2 committed:
added test of simpleParentSelector

AMazur2 committed:
fixed simpleParentSelector - out of range problem

AMazur2 committed:
added test_Recombinator
fixed out of range problems in singlePointRecombinator and twoPointRecombination

HasuNoHana commented on src/evolutionaryAlgorithm/EvolutionSimulator.py:
```diff
@@ -11,8 +11,8 @@
 class EvolutionSimulator:
-    initializator: InitializatorInterface
-    recombinator: RecombinatorInterface
+    initializator = InitializatorInterface
+    recombinator = RecombinatorInterface
```
WTF?

AMazur2 replied:
nie, nie to jest tak jak ty zrobiłaś a w jednym momencie się "bawiłe" bo nie kompilowało się a szukałem problemów więc nevermind

HasuNoHana commented on src/evolutionaryAlgorithm/EvolutionSimulator.py:
```diff
@@ -52,11 +53,19 @@ def run(self):
 
-    def selectParents(self, population: List[IndividualInterface]) -> List[IndividualInterface]:
-        return self.parentSelector.marry(population)
+    def selectParents(self, population: List[IndividualInterface], fitness: List[float]) -> List[IndividualInterface]:
+        return self.parentSelector.getParents(population, fitness)
```
Dlaczego tu dalej jest fitness

AMazur2 replied:
bo jeszcze nie ma zrobionej oceny, jak będzie ona zrobiona to szybko to zmienię

HasuNoHana commented on src/evolutionaryAlgorithm/SimulationComponents/ParentSelector/ParentSelectorInterface.py:
```diff
@@ -7,5 +7,5 @@
     
     @abstractmethod
-    def marry(self, population: List[IndividualInterface]):
+    def getParents(self, population: List[IndividualInterface], fitness: List[float]):
```
Tu też fitness ako argument, powinno być w individual

HasuNoHana commented on src/evolutionaryAlgorithm/SimulationComponents/ParentSelector/impl/simpleParentSelector.py:
```diff
@@ -11,13 +11,14 @@ class simpleParentSelector(ParentSelectorInterface):
         pass
     
-    def marry(self, Population: List[IndividualInterface]):
+    def getParents(self, population: List[IndividualInterface], fitness: List[float]) \
+            -> List[List[IndividualInterface]]:
```
Tu też lista fitnessów

HasuNoHana commented on src/evolutionaryAlgorithm/SimulationComponents/Recombinator/impl/averageRecombinator.py:
```diff
@@ -16,7 +16,7 @@ def subcross(self, marriage, w):
         return temp
 
-    def recombine(self, marriages: List):
+    def recombine(self, marriages: List) -> List[IndividualInterface]:
```
marrige jest listą czego?

AMazur2 replied:
Marriage : List[List[IndividualInterface]] w najnowszej wersji już jest zrobione tak

HasuNoHana commented on test/evolutionaryAlgorithm/SimulationComponents/Recombinator/impl/test_Recombinator.py:
```diff
@@ -0,0 +1,34 @@
+
+
+def test_Recombinator():
+    marriages = [[[1, 2, 3, 4], [5, 6, 7, 8]]]
```
To nie jest lista list individual interface tylko lista floatów!

AMazur2 replied:
Zmienione już

AMazur2 committed:
fixed problems with IndividualInterfaces in Recombinators
test_Recombinator now works on IndividualInterfaces (FLoatingPoint)

AMazur2 committed:
Recombinators now return a FloatingPointIndividual type
